### DISCUSSION TITLE
Primary Key Sync for Multiple Clients upload only

### DISCUSSION THREAD

**Discussion by rhemasystems** [Mimetis/Dotmim.Sync#1099] [Q&A]
Please how do you handle this case where am syncing one db to the server db, the server db already has 2 records in `Sales` table with primary keys 1 and 2 from another Client DB1, the new upload from the Client DB2 also has primary key 1 and 2, after syncing the  Client DB2, the upload goes to update the two records at the server db belonging to Client DB1 instead of inserting those 2 records making a total of 4 records, 2  records from each Client DB

Not that my Id field is `Identity Auto increment Primary keys` on both clients and server
Client DB2
![image](https://github.com/Mimetis/Dotmim.Sync/assets/50826372/52e8cb78-0f68-40db-8648-8c1f6615307b)

Server DB, showing the 2  records belonging to Client DB1 that have been updated by Client DB2 upload
![image](https://github.com/Mimetis/Dotmim.Sync/assets/50826372/9b68133d-d3e3-4504-a330-12af2d00030b)

From the screenshot, the new client db has updated the records in the server db instead of inserting.
and this table `Sales`  also has 2 related tables  called `SalesDetails ` and `SalesPayment `

**Comment by Mimetis** [comment]
that's intended :)
Primary Key 1 on any client will do an update on Server if Primary Key 1 is exsting already on the server

Just change Auto Increment to Guid

**Reply by rhemasystems** [reply]
Thank you sir, please what exactly do you mean.
Which datatype should I use.
Also what will be the impact on the related tables that have foreign key of this `Sales` table.

**Reply by Mimetis** [reply]
Change your primary keys and replace int `Identity Auto Increment` with `UniqueIdentifier` (guid)

**Reply by rhemasystems** [reply]
Thank you Sir,
I get the below error when I try to change the datatype

![image](https://github.com/Mimetis/Dotmim.Sync/assets/50826372/66e50011-6dc6-447f-8f84-9c7e08c97ac6)

**Reply by Mimetis** [reply]
https://learn.microsoft.com/en-us/sql/?view=sql-server-ver16
